### ISSUE TITLE
训练OCR识别模型ACC指标上不去

### ISSUE THREAD

**Issue by account15222** [PaddlePaddle/Paddle#75175]
### 请提出你的问题 Please ask your question

哪位大佬帮我分析一下这个问题呢？训练经文（类似传统蒙古文）时acc一直上不去，数据没问题
这个是配置文件 Global:
  debug: false
  use_gpu: true
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/v3_ganzhur
  save_epoch_step: 3
  eval_batch_step: [0, 500]
  cal_metric_during_train: true
  pretrained_model: ./output/v3_ganzhueri/best_accuracy
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ppocr/utils/dict/ganzhur_mong_dict.txt
  max_text_length: &max_text_length 60
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ganzhur_ppocrv3.txt

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0005
    warmup_epoch: 5
  regularizer:
    name: L2
    factor: 3.0e-05

Architecture:
  model_type: rec
  algorithm: SVTR
  Transform:
  Backbone:
    name: MobileNetV1Enhance
    scale: 0.5
    last_conv_stride: [1, 2]
    last_pool_type: avg
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 64
            depth: 2
            hidden_dims: 120
            use_guide: True
          Head:
            fc_decay: 0.00001
      - SARHead:
          enc_dim: 512
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - SARLoss:

PostProcess:  
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc
  ignore_space: False
Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/rec
    ext_op_transform_idx: 1
    label_file_list:
    - ./train_data/rec/ganzhur_train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
   # - RecConAug:
   #     prob: 0.3
   #     ext_data_num: 2
   #     image_shape: [48, 320, 3]   # 改大，避免长文本截断
   # - RecAug:
    - MultiLabelEncode:
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_sar
        - length
        - valid_ratio
  loader:
    shuffle: true
    batch_size_per_card: 128
    drop_last: true
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/rec
    label_file_list:
    - ./train_data/rec/ganzhur_test.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_sar
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 128
    num_workers: 4

日志是[2025/09/09 11:14:18] ppocr INFO: epoch: [12/500], global_step: 2560, lr: 0.000500, acc: 0.125000, norm_edit_dis: 0.724426, CTCLoss: 0.919947, SARLoss: 0.333542, loss: 1.255447, avg_reader_cost: 0.18238 s, avg_batch_cost: 1.06900 s, avg_samples: 102.4, ips: 95.79060 samples/s, eta: 1 day, 5:35:00
[2025/09/09 11:14:30] ppocr INFO: epoch: [12/500], global_step: 2570, lr: 0.000500, acc: 0.128906, norm_edit_dis: 0.733763, CTCLoss: 0.922717, SARLoss: 0.330066, loss: 1.247488, avg_reader_cost: 0.00041 s, avg_batch_cost: 1.15004 s, avg_samples: 128.0, ips: 111.30045 samples/s, eta: 1 day, 5:36:23
[2025/09/09 11:14:41] ppocr INFO: epoch: [12/500], global_step: 2580, lr: 0.000500, acc: 0.121094, norm_edit_dis: 0.734781, CTCLoss: 1.068798, SARLoss: 0.330193, loss: 1.390958, avg_reader_cost: 0.00086 s, avg_batch_cost: 1.13902 s, avg_samples: 128.0, ips: 112.37767 samples/s, eta: 1 day, 5:37:42
[2025/09/09 11:14:52] ppocr INFO: epoch: [12/500], global_step: 2590, lr: 0.000500, acc: 0.117187, norm_edit_dis: 0.726507, CTCLoss: 1.107466, SARLoss: 0.334788, loss: 1.455399, avg_reader_cost: 0.00035 s, avg_batch_cost: 1.05149 s, avg_samples: 128.0, ips: 121.73151 samples/s, eta: 1 day, 5:38:21
[2025/09/09 11:15:01] ppocr INFO: epoch: [12/500], global_step: 2600, lr: 0.000500, acc: 0.125000, norm_edit_dis: 0.724565, CTCLoss: 1.011954, SARLoss: 0.335290, loss: 1.336439, avg_reader_cost: 0.00058 s, avg_batch_cost: 0.94621 s, avg_samples: 128.0, ips: 135.27653 samples/s, eta: 1 day, 5:38:14
[2025/09/09 11:15:11] ppocr INFO: epoch: [12/500], global_step: 2610, lr: 0.000500, acc: 0.128906, norm_edit_dis: 0.724301, CTCLoss: 1.013325, SARLoss: 0.334609, loss: 1.339103, avg_reader_cost: 0.00059 s, avg_batch_cost: 0.94802 s, avg_samples: 128.0, ips: 135.01772 samples/s, eta: 1 day, 5:38:08
[2025/09/09 11:15:20] ppocr INFO: epoch: [12/500], global_step: 2620, lr: 0.000500, acc: 0.113281, norm_edit_dis: 0.730116, CTCLoss: 0.973506, SARLoss: 0.336636, loss: 1.319104, avg_reader_cost: 0.00056 s, avg_batch_cost: 0.96190 s, avg_samples: 128.0, ips: 133.06971 samples/s, eta: 1 day, 5:38:07
[2025/09/09 11:15:30] ppocr INFO: epoch: [12/500], global_step: 2630, lr: 0.000500, acc: 0.113281, norm_edit_dis: 0.730862, CTCLoss: 0.942175, SARLoss: 0.337761, loss: 1.282220, avg_reader_cost: 0.00032 s, avg_batch_cost: 0.98287 s, avg_samples: 128.0, ips: 130.23028 samples/s, eta: 1 day, 5:38:16

**Comment by zhangyubo0722** [comment]
训练到多少epoch发现精度上不去呢

**Comment by hjiehudbandbegshth** [comment]
可能是因为CTCHead + SARHead 混在一起训练，但后处理和 metric 又只用了 CTC ，导致两个 head 都无法正常收敛，acc 一直卡在 0.11～0.13 左右。

在 PaddleOCR 里：

CTC 系列（CTCHead）：CTCLoss + CTCLabelDecode + RecMetric

SAR 系列（SARHead）：SARLoss + SARLabelDecode + RecMetric

两套体系的解码方式完全不同，不能混用。现在的配置等于是让模型在训练两种 head，但只评估 CTC，SARHead 在训练中完全“瞎跑”，还会干扰 CTC 的 loss
只采用CTC试试呢
配置文件：
Head: CTCHead
Loss: CTCLoss
PostProcess: CTCLabelDecode
Metric: RecMetric
